scripts/propose.py

```diff
@@ -1,16 +1,20 @@
 import grpc
 import logging
+import sys
 from rchain.client import RClient
 from rchain.client import RClientException
 
 root = logging.getLogger()
-root.setLevel(logging.INFO)
+root.setLevel(logging.DEBUG)
 
+ret=0
 with grpc.insecure_channel('localhost:40401') as channel:
     client = RClient(channel)
     try:
         client.propose()
     except RClientException as e:
-        # NoNewDeploys is a valid scenario
+        # NoNewDeploys case should not exit with an error \
+        # so autopropose script can deploy again before next propose
         if "NoNewDeploys" not in str(e):
-            raise
+            ret=1
+sys.exit(ret)
```

Comment:
When _main()_ raises an exception, the interpreter exists with non-zero exit code by default. Furthermore, with this construct, you effectively silence the exception output.